05_stringer/starfallz/main.go

```diff
@@ -0,0 +1,27 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"strings"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	fmt.Fprint(out, IPAddr{127, 0, 0, 1})
+}
+
+type IPAddr struct {
+	add1, add2, add3, add4 uint8
```

Comment:
uint8 is correct that it gives the correct IP Address range of 0-255, and the clue given by the [lecture notes](https://tour.golang.org/methods/18) is using array of 4 bytes (type IPAddr [4]byte). From a technical perspective, it is more accurate, as IPv4 address is represented as bytes( 8 bits). My guess is that the program would run faster using byte as it more native at the hardware level. I actually googled around, and seems mos t implementations in go is implemented as Byte and not uint8.

Reply:
If I'm not wrong byte is an alias of uint8 and uint8 is a more explicit way of declaring it. But at the end of the day it really depends on people's preferences interms of readability. I am more than happy to change it to byte but I do want to hear from any of the course reviewer about it though. Maybe you are right since in this case it's an ip address, maybe byte alias is more relevant.

Reply:
Personally I don't have problems with uint32, this is more of a comment to spark some conversations. I believe that the data type should reflect the type of data that you'll be getting, and in this case. it will be bytes. I do think that having array instead of 4 different variables will help, your String() will just be a one line using string methods. So this is something you can explore.

Reply:
I actually don't have a problem with it either way. Golang inlines structs instances, so this will be just as efficient space and time-wise as `[4]byte`. The difference is how you use the `IPAddr` type.
`[4]byte` gives you array access and the ability to range over the elements of the address.
your `struct` allows you to only initialize say the third element (use `IPAddr{add3: 1}`).

My preference is towards `[4]byte`, but its not a strong preference.

Reply:
Additionally, @starfallz is correct in saying `byte` is an alias of `uint8`. There really is no difference between the two, only in that bytes are more often used for text based data handling.

Reply:
Please change as per reviewer requests (it will test cases nicer too).